linter/resource_linter.go

```diff
@@ -23,7 +23,15 @@ func (r ResourceLinter) ValidateResources(resources []assertion.Resource, rules
 
 	for _, rule := range resolvedRules {
 		assertion.Debugf("Rule: ID: %v Message: %s\n", rule.ID, rule.Message)
-		for _, resource := range assertion.FilterResourcesByType(resources, rule.Resource, rule.Category) {
+		var filteredResources []assertion.Resource
+		if rule.Resource != "" {
+			assertion.Debugf("using the rule - string resource")
+			filteredResources = assertion.FilterResourcesByType(resources, rule.Resource, rule.Category)
+		} else {
+			assertion.Debugf("using the rule - list of resources")
+			filteredResources = assertion.FilterResourcesByTypes(resources, rule.Resources, rule.Category)
+		}
+		for _, resource := range filteredResources {
```

Comment:
I would recommend extracting these lines into a new filter function in assertion/util.go. Maybe call it FilterResourcesForRule? Pass the list of Resource objects and the entire Rule object. That would keep this loop code more concise, and groups all the filtering logic together in the same package. What do you think?